### PR TITLE
add Chinese traditional version for Taiwan

### PR DESCRIPTION
**Pull request by tso1158687**
add chinese traditional version for taiwan
this translation depends on

- translation on angular.tw
- change link from angular.io to angular.tw

### CODE REVIEW

File: src/app/locales/zh_TW.ts [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+    action: '從 Angular 9 開始，對 DI 的 @Injectable 裝飾器執行更加嚴格，不完整的提供者程序定義的行為也有所不同。<a href="https://angular.tw/guide/migration-injectable" target="_blank">了解更多</a>。ng update 將自動搬移。'
+  },
+  'closure-jsdoc-comments': {
+    action: 'Angular 的 NPM 套件不再包含 jsdoc 註解，這是與閉包編譯器一起使用所必須的(非常罕見)。此支援是實驗性的，且只在某些使用案例中有效。不久後將會宣布一條替代的推薦路線。'
+  },
+  'forms-number-input': {
+    action: '如果你使用 Angular 表單，則 `number` 型別的輸入將不再監聽<a href="https://developer.mozilla.org/en-us/docs/Web/API/HTMLElement/change_event" target="_blank">更改事件</a>(此事件不必為每個更改觸發該值)，而監聽<a href="https://developer.mozilla.org/en-us/docs/Web/API/HTMLElement/input_event" target="_blank">輸入事件</a>。'
+  },
+  'forms-length-input': {
+    action: '對於 Angular 表單驗證，`minLength` 和 `maxLength` 驗證器現在在正表單控制組件具有數字長度的屬性，且唯有在此情況下才會驗證長度。'
```
Comment (tomyjwu) [inline src/app/locales/zh_TW.ts:329]: 對於 Angular 表單驗證，表單控件的值的長度 (length) 屬性是數字類別時，`minLength` 和 `maxLength` 驗證器才會對此控件做長度驗證。

**Review by tomyjwu — COMMENTED** [review]
翻譯是很主觀的，僅提供一些意見供參考。

**Comment by mgechev** [comment]
@tso1158687 would you have a look at the comments by @tomyjwu? We can merge once you address them :)

File: src/app/locales/zh_TW.ts [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+    action: '`keyvalue` 管道已經修正，可以報告具有數字作為 key 的輸入物件，結果的型別將包含鍵的字串表示形式。現在已經是這樣，並且僅對程式碼進行更新來反映這點。如果相依於不正確的型別，請更新管道輸出的使用者。請注意，這不會影響輸入值為 `Map` 的使用案例，因此，如果需要保留 `number`，這是一個有效的方法。'
+  },
+  'v11 number pipe typing': {
+    action: '數字管道(`decimal`， `percent`， `currency`等)管道現在明確宣告接受哪些類別。'
+  },
+  'v11 date pipe typing': {
+    action: '`date` 管道現在明確宣告接受哪些類別。'
+  },
+  'v11 datetime rounding': {
+    action: '當將時間日期包含毫秒的字串的格式傳送給 `DatePipe` 時，現在毫秒將會四捨入，而非取最接近的毫秒。大多數的應用程式將會不受此更改影響。如果這不是你預期的行為，那麼考慮對字串先行處理四捨五入至毫秒再傳送給 `DatePipe`。'
```
[the remark this answers is not in the document]
Comment (tso1158687) [reply src/app/locales/zh_TW.ts:416]: 已修正，打到昏頭了

File: src/app/locales/zh_TW.ts [hunk 3]
```diff
@@ -0,0 +1,444 @@
...
+  },
+  'v11 async pipe typing': {
+    action: '`async` 管道不再要求輸入為 undefined 的輸入返回 undefined。要注意的是，程式碼實際上在位定義的輸入返回 null。'
```
[the remark this answers is not in the document]
Comment (tso1158687) [reply src/app/locales/zh_TW.ts:419]: 已修正錯字

File: src/app/locales/zh_TW.ts (hunk 1 above)
Comment (tso1158687) [reply src/app/locales/zh_TW.ts:329]: 謝謝，當初想了好久，這樣翻譯順多了

File: src/app/locales/zh_TW.ts [hunk 4]
```diff
@@ -0,0 +1,444 @@
...
+    action: '將你的 `template` 標籤重新命名為 `ng-template`'
+  },
+  'OpaqueToken': {
+    action: '使用 `InjectionToken` 取代 `OpaqueToken`。'
+  },
+  'DifferFactory': {
+    action: '如果呼叫 `DifferFactory.create(...)`，移除 `ChangeDetectorRef` 參數。'
+  },
+  'ErrorHandler Parameter': {
+    action: '停止傳遞任何參數給 ErrorHandler 的建構式'
```
[the remark this answers is not in the document]
Comment (tso1158687) [reply src/app/locales/zh_TW.ts:97]: 既然在angular.tw都稱為建構函式，那麼就以angular.tw的翻譯為準吧!

File: src/app/locales/zh_TW.ts [hunk 5]
```diff
@@ -0,0 +1,444 @@
...
+    action:
+      '停止使用 `Renderer.invokeElementMethod`，因為此方法已經移除了，目前没有替代方法。'
+  },
+  'Non Animations Module': {
+    action:
+      '如果你在應用程式中使用動畫，你應該在 `NgModule` 匯入 `@angular/platform-browser/animations` 中的 `BrowserAnimationsModule`。'
+  },
+  'Native Form Validation': {
+    action:
+      '当你包含 `FormsModule` 时，Angular 开始向表单元素添加一个 `novalidate` 属性，重新启用原始(Native)表单验证使用 `ngNoForm` 或添加 `ngNativeValidate`。'
```
[the remark this answers is not in the document]
Comment (tso1158687) [reply src/app/locales/zh_TW.ts:69]: 謝謝，應該是校對的時候蓋掉原本的了

File: src/app/locales/zh_TW.ts [hunk 6]
```diff
@@ -0,0 +1,444 @@
...
+  'From': '從',
+  'To': '到',
+  'Warning': '注意',
+  'Plans for releases after the current major release are not finalized and may change. These recommendations are based on scheduled deprecations.':
+    '目前主要版本後的規劃尚未定案，且可能有所變更。這些建議基於已排定的棄用。',
+  'We do not currently support downgrading versions of Angular.':
+    '現在並不支援 Angular 版本降級。',
+  'We do not recommend moving across multiple major versions.':
+    '不建議升級時跨越太多主要版本。',
+  'App Complexity': '程式複雜度',
```
[the remark this answers is not in the document]
Comment (tso1158687) [reply src/app/locales/zh_TW.ts:18]: 統一稱為應用程式

File: src/app/locales/zh_TW.ts [hunk 7]
```diff
@@ -0,0 +1,444 @@
...
+  'Show all the information we have about the update.': '顯示所有更新資訊。',
+  'Other Dependencies': '其他相依',
+  'I use': '使用',
+  'to combine AngularJS & Angular': '結合 AngularJS 和 Angular',
+  'Package Manager': '套件管理',
+  'Show me how to update!': '顯示如何更新！',
+  'Before Updating': '更新前',
+  // tslint:disable-next-line: quotemark
+  "There aren't currently any changes needed before moving between these versions.":
+    '目前在这两个版本间变化无需任何变更目前這兩個版本間的變化。',
```
[the remark this answers is not in the document]
Comment (tso1158687) [reply src/app/locales/zh_TW.ts:35]: 謝謝，應該是校對的時候不小心貼到了

File: src/app/locales/zh_TW.ts [hunk 8]
```diff
@@ -0,0 +1,444 @@
...
+    action: '如果使用 TrackByFn，請改用 TrackByFunction'
+  },
+  'i18n Pipe Change': {
+    action: '如果相依於日期(date)，貨幣(currency)，十進位(decimal)或百分比(percent)管道，則將在 5 中會看見格式的細微變化。對於使用英語(en-us)以外的應用程式，需要從 `@angular/common/i18n_data/locale_fr` 匯入 `locale` 並使用 `registerLocaleData(local)` 註冊本地語言資料。'
+  },
+  'gendir': {
+    action: '切勿依賴 `gendir`，而考慮使用 `skipTemplateCodeGen`。<a href="https://github.com/angular/angular/issues/19339#issuecomment-332607471" target="_blank">了解詳情</a>'
+  },
+  'Dynamic ngUpgrade': {
+    action: '替换从 `@angular/upgrade` 导入的 `downgradeComponent`、`downgradeInjectable`、`UpgradeComponent` 和 `UpgradeModule`。需要在 `@angular/upgrade/static` 中使用新版本'
```
[the remark this answers is not in the document]
Comment (tso1158687) [reply src/app/locales/zh_TW.ts:112]: 謝謝，應該在校對的時候貼到了

File: src/app/locales/zh_TW.ts [hunk 9]
```diff
@@ -0,0 +1,444 @@
...
+  'We do not currently support downgrading versions of Angular.':
+    '現在並不支援 Angular 版本降級。',
+  'We do not recommend moving across multiple major versions.':
+    '不建議升級時跨越太多主要版本。',
+  'App Complexity': '程式複雜度',
+  'Basic': '基本',
+  'Medium': '中等',
+  'Advanced': '進階',
+  'Show update information relevant to all Angular developers.':
+    '顯示所有與Angular 開發人員相關更新資訊。',
```
[the remark this answers is not in the document]
Comment (tso1158687) [reply src/app/locales/zh_TW.ts:23]: 顯示 Angular 開發人員都要知道的更新資訊。

File: src/app/locales/zh_TW.ts [hunk 10]
```diff
@@ -0,0 +1,444 @@
...
+  'Show more specific update information relevant to around 10% of Angular developers.':
+    '顯示大約關於10% Angular 開發人員相關詳細更新資訊。',
```
[the remark this answers is not in the document]
Comment (tso1158687) [reply src/app/locales/zh_TW.ts:25]: 顯示約前10% Angular 開發人員詳細更新資訊。

File: src/app/locales/zh_TW.ts [hunk 11]
```diff
@@ -0,0 +1,444 @@
...
+    action: '`async` 管道不再要求輸入為 undefined 的輸入返回 undefined。要注意的是，程式碼實際上在位定義的輸入返回 null。'
+  },
+  'v11 case pipe update': {
+    action: '`uppercase` 和 `lowercase` 管道不再讓假值通過。現在將會將 `null` 和 `undefined` 都映射到 `null`，並且在無效的輸入(`0`， `false`， `NaN`)上引發例外錯誤。和其他 Angular 管道一致。'
+  },
+  'v11 router NavigationExtras typing': {
+    action: '如果將路由(Router)與 `NavigationExtras` 一起使用，則新的型別允許傳入型別為 `NavigationExtras` 的變數，但不允許實字物件(object literals)，因为可能僅指定已知屬性。同時也不接受那些與 `Pick` 中的屬性沒有共同屬性的型別。如果受到此更改的影響，請僅從 NavigationExtras 中指定實際在各個函式呼叫中使用的屬性，或在物件或變數上使用型別聲明：`as NavigationExtras`。'
+  },
+  'v11 TestBed.overrideProvider': {
+    action: '在測試中，如果在 TestBed 初始化後呼叫 `TestBed.overrideProvider`，overrideProvider 已移除。此行為與其他替代方法(例如 `TestBed.overrideDirective` 等)一致，但會拋出錯誤指出這點。此檢查以前在 TestBed.overrideProvider 函數中錯失。如果看到該錯誤，應該在 TestBed 初始化之前移動 `TestBed.overrideProvider` 呼叫。'
```
[the remark this answers is not in the document]
Comment (tso1158687) [reply src/app/locales/zh_TW.ts:428]: 謝謝這樣好像比較順

**Comment by tso1158687** [comment]
@googlebot I signed it!

**Comment by tso1158687** [comment]
@googlebot I fixed it.

**Comment by mgechev** [comment]
@tso1158687 looks like the CLA bot is unhappy because of [this](https://github.com/angular/angular-update-guide/pull/108/commits/45edbb06d3549f57f7e9ef7bb43ad77d7a4b887c) commit. If it's part of your contribution you can squash it as part of another commit, or drop it alternatively.

After that we should be able to merge your PR.

Please, directly mention me once you're done to make sure I don't miss the notification. :)

**Comment by tso1158687** [comment]
Hi @mgechev,
I have squashed and forced push the commit and all checks have passed.

Please help me to merge this commit into master.

Thank you!

**Comment by doggy8088** [comment]
Hi @tso1158687 , could you take a look of my reviews? There are some words needs to be revised.

**Review by doggy8088 — CHANGES_REQUESTED** [review]

**Comment by tomyjwu** [comment]
> @mgechev sure!
> I have updated translation by @tomyjwu advices

thanks. changes looks to me . +1

**Comment by tomyjwu** [comment]
@tso1158687 請再確認一下保哥的comments 即可。

**Review by doggy8088 — APPROVED** [review]
LGTM

**Comment by tso1158687** [comment]
@tomyjwu @doggy8088  yes I resolved all comments in this [commit](https://github.com/angular/angular-update-guide/pull/108/commits/832a61de4a93110e007e61f0fe1a87005321a54d)

@mgechev please help me to merge into master.
Thank you all!

**Review by tomyjwu — APPROVED** [review]
LGTM

**Review by mgechev — APPROVED** [review]

**Comment by mgechev** [comment]
Thank you for the translation!

**Pull request merged**